draft-ietf-jsonpath-base.md

```diff
@@ -902,11 +965,173 @@ size doesn't
 fit in the implementation's representation of an integer, the implementation
 MUST raise an error.
 
+### Descendant Selector
+
+#### Syntax
+{: unnumbered}
+
+The descendant selector starts with a double dot `..` and can be
+followed by an object member name (similar to the `dot-selector`),
+by an `index-selector` acting on objects or arrays, or by a wild card.
+
+~~~~ abnf
+descendant-selector = ".." ( dot-member-name      /  ; ..<name>
+                             index-selector       /  ; ..[<index>]
+                             index-wild-selector  /  ; ..[*]
+                             "*"                     ; ..*
+                           )
+~~~~
+
+#### Semantics
+{: unnumbered}
+
+The `descendant-selector` is inspired by ECMAScript for XML (E4X). It
+selects the nodes of object member values and array elements in a recursive descent process.
+
+### Union Selector
+
+#### Syntax
+
+The union selector is syntactically related to the `index-selector`. It contains multiple, comma separated entries.
 
+~~~~ abnf
+union-selector = "[" ws union-entry 1*(ws "," ws union-entry) ws "]"
 
+union-entry    =  ( quoted-member-name /
+                    element-index      /
+                    slice-index
+                  )
+~~~~
+
+> Task (T1): This, besides slice-index, is currently one of only two
+> places in the document that mentions whitespace.  Whitespace needs
+> to be handled throughout the ABNF syntax.  Room Consensus at the
+> 2021-06-15 interim was that JSONPath generally is generous with
+> allowing insignificant whitespace throughout.  Minimizing the impact
+> of the many whitespace insertion points by choosing a rule name such
+> as "S" was mentioned.  Some conventions will probably help with
+> minimizing the number of places where S needs to be inserted.
```

Comment:
For the record, "ws" is far more memorable than "S".

Reply:
If you are stuck in an ABNF littered with them, S is sufficiently memorable...

(WS really should be for mandatory whitespace, OWS is optional whitespace, but that is too long here.)

Reply:
I never got that color thing people associate with space - maybe I should buy me a printer and do some archaeology? Just  `S` or `SP` should do nicely.

Reply:
I am fine with "S" (minimalistic :). Is there a convention to write constants in capitals?

Reply:
Do we need to make concession for multiple whitespace chars (e.g. `*ws` or `*S`), or is that implicitly covered?

Reply:
RFC 8610 defines this as:

     S = *WS
     WS = SP / NL
     SP = %x20
     NL = COMMENT / CRLF
     COMMENT = ";" *PCHAR CRLF
     PCHAR = %x20-7E / %x80-10FFFD
     CRLF = %x0A / %x0D.0A

I'm not sure we need to support comments in JSONPath (that is a serious question though), but if you remove that, pesky details like isolated CRs are fixed, too (and then CDDL simply disallows HTs, which may not be popular here).

Next PR...

Reply:
Oh, and I'm not sure these can all be called constants, but there is indeed a tendency in ABNF usage to give token-level (scanner) productions such as DIGIT, ALPHA, WS, ... upper case names.